src/MongoDB.Driver.Core/Core/Servers/ServerDescription.cs

```diff
@@ -472,6 +472,28 @@ public override int GetHashCode()
                 .GetHashCode();
         }
 
+        /// <summary>
+        /// Determines whether the specified <see cref="ServerDescription" /> can be considered as equal to decide should we publish sdam events or not.
+        /// </summary>
+        /// <param name="other">The other server description.</param>
+        /// <returns><c>true</c>, if sdam events should be suppressed, otherwise <c>false</c>.</returns>
+        public bool SdamEquals(ServerDescription other)
+        {
+            var canBeConsideredAsEqual = 
```

Comment:
Perhaps name the variable `sdamEquals`.

Or better yet... remove the variable and return directly?